libopenage/renderer/camera/camera.h

```diff
@@ -84,7 +85,9 @@ class Camera {
 	 *
 	 * @param scene_pos New 3D position of the camera in the scene.
 	 */
-	void move_to(Eigen::Vector3f scene_pos);
+	void move_to(Eigen::Vector3f scene_pos, std::optional<std::pair<float, float>> x_bounds = std::nullopt,
+											std::optional<std::pair<float, float>> z_bounds = std::nullopt);
```

Comment:
Instead of using `std::optional` and assigning `std::nullopt` as default value, you could also use regular non-optional floats and set the default value to `std::numeric_limits::min()`/`std::numeric_limits::max()`. Then the clamp operation will work the same, but you won't need to check `has_value` in the code.